lib/Basics/FixedSizeAllocator.h

```diff
@@ -89,8 +97,8 @@ class FixedSizeAllocator {
    private:
     size_t _numAllocated;
     size_t _numUsed;
-    char* _alloc;
     char* _data;
+    std::unique_ptr<MemoryBlock> _next;
```

Comment:
While this works, I think it could create very deep stacks on destruction of the memory block.
If, for example, there is a chain of 100 memory blocks, the dtor of each block will invoke the dtor of the next block etc., so this will cause at least 100 stack frames to be produced.
IMHO this is not ideal, as it prevents scaling the number of MemoryBlocks.